var/spack/repos/builtin/packages/py-anuga/package.py

```diff
@@ -13,20 +13,28 @@ class PyAnuga(PythonPackage):
 
     homepage = "https://github.com/GeoscienceAustralia/anuga_core"
     url      = "https://github.com/GeoscienceAustralia/anuga_core/archive/2.1.tar.gz"
+    git      = 'https://github.com/GeoscienceAustralia/anuga_core.git'
 
+    # The git main branch of the repo is now python3-only
+    version('main', branch='main')
     version('2.1', sha256='0e56c4a7d55570d7b2c36fa9b53ee4e7b85f62be0b4c03ad8ab5f51464321d2f')
 
     variant('mpi', default=True, description='Install anuga_parallel')
 
     # At present AnuGA has only been run and tested using python 2.x.
     # We recommend python 2.7.
-    depends_on('python@2.6:2.8', type=('build', 'run'))
-    depends_on('py-setuptools@:44', type='build')
-    depends_on('py-numpy@:1.16', type=('build', 'run'))
+    depends_on('py-setuptools@45:',      when='@2.2:,main', type=('build'))
+    depends_on('py-setuptools@:44.1.0',  when='@:2.1',      type=('build'))
+    depends_on('py-setuptools-scm@:3.5', when='@:2.1',      type=('build'))
+    depends_on('python@3.5:',            when='@2.2:,main', type=('build', 'run'))
+    depends_on('python@2.6:2.7.18',      when='@:2.1',      type=('build', 'run'))
+    depends_on('py-numpy@1.17:',         when='@2.2:,main', type=('build', 'run'))
+    depends_on('py-numpy@:1.16.6',       when='@:2.1',      type=('build', 'run'))
```

Comment:
I don't see any documented requirements for setuptools version, usage of setuptools-scm, requirements for Python 3 version, or for numpy version. Where are these coming from?

Reply:
Updated to now keep only the version limitations given by the current package.py and a small, but described workaround:
```py
+    # Version-restricted dependencies for anuga@:2.1, converted to use when='@:2.1':
+    depends_on('python@2.6:2.7.18',      type=('build', 'run'), when='@:2.1')
+    depends_on('gdal@:3.2+geos+python',  type=('build', 'run'), when='@:2.1')
+    depends_on('py-matplotlib@:2',       type=('build', 'run'), when='@:2.1')
+    depends_on('py-numpy@:1.16',         type=('build', 'run'), when='@:2.1')
+    depends_on('py-setuptools@:44',      type=('build'),        when='@:2.1')
+    # Workaround for problem in the original concretizer selecting 3.6, wanting python3:
+    depends_on('py-setuptools-scm@:3.5', type=('build'),        when='@:2.1')
+    # pypar is not updated for python3, for python3, py-mpi4py is unsed instead:
+    depends_on('py-pypar',               type=('build', 'run'), when='@:2.1+mpi')
```